proxy/ProxyClientSession.cc

```diff
@@ -142,7 +141,9 @@ ProxyClientSession::do_api_callout(TSHttpHookID id)
   this->api_current = NULL;
 
   if (this->hooks_on && this->has_hooks()) {
-    SET_HANDLER(&ProxyClientSession::state_api_callout);
+    if (!this->handler) {
```

Comment:
I'm not sure about this. The state of the handler doesn't seem to be well-defined at this point. HTTP2 sets the handler in ``start()``, for example.